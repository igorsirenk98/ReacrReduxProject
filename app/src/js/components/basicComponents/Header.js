import React, { Component } from 'react';
import { connect } from 'react-redux';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import IconButton from '@material-ui/core/IconButton';
import Typography from '@material-ui/core/Typography';
import InputBase from '@material-ui/core/InputBase';
import { fade, makeStyles } from '@material-ui/core/styles';
import MenuIcon from '@material-ui/icons/Menu';
import SearchIcon from '@material-ui/icons/Search';

import { SearchInput } from './SearchInput';
import { fetchProductsBySearch } from '../../actions/ProductsBySearch';

const useStyles = makeStyles(theme => ({
    toolbar: {
      borderBottom: `1px solid ${theme.palette.divider}`,
    },
    toolbarTitle: {
      flex: 1,
    },
    toolbarSecondary: {
      justifyContent: 'space-between',
      overflowX: 'auto',
    },
    toolbarLink: {
      padding: theme.spacing(1),
      flexShrink: 0,
    },
    searchField: {
        color: '#fff'
    }
}));

export class Header extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <>
                <AppBar position="static">
                    <Toolbar>
                        <IconButton
                            edge="start"
                            color="inherit"
                            aria-label="open drawer"
                        >
                        <MenuIcon />
                        </IconButton>
                        <Typography variant="h6" noWrap>
                            Bike shop
                        </Typography>
                        <div>
                        <IconButton aria-label="search">
                            <SearchIcon />
                        </IconButton>
                        <SearchInput />
                        </div>
                    </Toolbar>
                </AppBar>
            </>
        )
    }
}