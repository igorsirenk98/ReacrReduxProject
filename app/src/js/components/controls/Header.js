import React from 'react';
import { Link } from 'react-router-dom';

import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import Tooltip from '@material-ui/core/Tooltip';
import DirectionsBikeIcon from '@material-ui/icons/DirectionsBike';

import SearchInput from '../../containers/controls/SearchInput/SearchInput';

const Header = () => (
    <>
        <AppBar className="appbar" position="static">
            <Toolbar className="toolbar">
                <Typography variant="h6" noWrap>
                    <Tooltip title="Go to main page" arrow>
                        <Button>
                            <Link
                                to={{ pathname: '/' }}
                                className="link"
                            >
                                Bike shop
                                <DirectionsBikeIcon />
                            </Link>
                        </Button>
                    </Tooltip>
                </Typography>
                <div>
                    <SearchInput />
                </div>
            </Toolbar>
        </AppBar>
    </>
);

export default Header;
