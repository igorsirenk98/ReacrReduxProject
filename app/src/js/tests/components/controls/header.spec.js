import React from 'react'
import { shallow } from 'enzyme'
import Header from '../../../components/controls/Header';

describe('Header component', () => {
    const header = shallow(<Header />);

    it('renders correctly', () => {
        expect(header.debug()).toMatchSnapshot()
    })
})
