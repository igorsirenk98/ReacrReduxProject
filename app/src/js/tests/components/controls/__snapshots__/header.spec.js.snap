// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly 1`] = `
"<Fragment>
  <WithStyles(ForwardRef(AppBar)) className=\\"appbar\\" position=\\"static\\">
    <WithStyles(ForwardRef(Toolbar)) className=\\"toolbar\\">
      <WithStyles(ForwardRef(Typography)) variant=\\"h6\\" noWrap={true}>
        <WithStyles(ForwardRef(Tooltip)) title=\\"Go to main page\\" arrow={true}>
          <WithStyles(ForwardRef(Button))>
            <Link to={{...}} className=\\"link\\">
              Bike shop
              <DirectionsBikeIcon />
            </Link>
          </WithStyles(ForwardRef(Button))>
        </WithStyles(ForwardRef(Tooltip))>
      </WithStyles(ForwardRef(Typography))>
      <div>
        <Connect(SearchInput) />
      </div>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
</Fragment>"
`;
